core/orchestrator.go

```diff
@@ -102,26 +102,43 @@ func (orch *orchestrator) ProcessPayment(payment net.Payment, manifestID Manifes
 		return nil
 	}
 
-	ticket := &pm.Ticket{
-		Recipient:         ethcommon.BytesToAddress(payment.Ticket.Recipient),
-		Sender:            ethcommon.BytesToAddress(payment.Ticket.Sender),
-		FaceValue:         new(big.Int).SetBytes(payment.Ticket.FaceValue),
-		WinProb:           new(big.Int).SetBytes(payment.Ticket.WinProb),
-		SenderNonce:       payment.Ticket.SenderNonce,
-		RecipientRandHash: ethcommon.BytesToHash(payment.Ticket.RecipientRandHash),
+	if payment.TicketParams == nil {
+		return errors.Wrapf(errors.New("No TicketParams in payment"), "Could not find TicketParams for payment %v", payment)
```

Comment:
`errors.Wrapf` is typically used when:

- Function A calls function B calls function C
- C returns an error to B
- B wraps C's error with its own error
- A receives the wrapped error which includes context from both function B and C

We're not receiving an error from a function call here so we can just return a single error instead of wrapping an error with another error.